test/contrib/external_program_test.py

```diff
@@ -184,6 +184,44 @@ def test_non_mocked_task_run(self):
             # clean up temp files even if assertion fails
             shutil.rmtree(tempdir)
 
+    def test_tracking_url_pattern_works_with_capture_output_disabled(self):
+        temp_dir = tempfile.mkdtemp()
+
+        # since set_tracking_url is called in a separate process
+        # we need to do something like creating a temp file
```

Comment:
Maybe you want https://docs.python.org/3/library/multiprocessing.html#sharing-state-between-processes ?

Reply:
Thank you, changed, now looks much better